pkg/authentication/tokens.go

```diff
@@ -0,0 +1,64 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+	"github.com/golang-jwt/jwt/v5"
+	"strings"
+)
+
+// auth0 is an Authentication implementation using Auth0 as an authentication provider.
+type auth0 struct {
+	publicKey []byte
+}
+
+// VerifyJWT verifies that the given token is a valid JWT and was correctly signed by Auth0.
+func (auth *auth0) VerifyJWT(ctx context.Context, token string) error {
+	if err := auth.validateToken(token); err != nil {
+		return err
+	}
+	parsedToken, err := jwt.Parse(token, auth.keyFunc)
+	if err != nil {
+		return err
+	}
+	if !parsedToken.Valid {
+		return errors.New("token is not valid")
```

Comment:
We should define this and any other common errors in the generic `authentication` package alongside the `Authorization` interface, so that all implementations can use it and user error handling can be made independent of implementation.

Wrapping the generic error to provide more context is fine, the main objective here is for end users to have a common errors to check for and react to when calling `auth` methods.